### PR TITLE
Tests and files for creators

### PR DESCRIPTION
**Pull request by urbantumbleweed**
#### What's this PR do?

This adds a new directory for action creators.  Code comprising the creators logic was moved out of their corresponding actions files into their own files.  Action creators for errors were created where needed.  In the creators directory is a '**test**' directory where all the creator tests are located.
#### What are the important parts of the code?

For an overview of the code start with the index.js file in creators directory.
#### How should this be tested by the reviewer?

`git pull origin creators && npm install && npm run test`
#### Is any other information necessary to understand this?

creators are used within action files.  The action determines whether there is only a success case or if there is also an error case.

### CODE REVIEW

**Comment by urbantumbleweed** [comment]
I refactored the code to incorporate the changes you wanted.  Most notably is the removal of many beforeEach and afterEach blocks because the tests don't change anything.  i still need to write tests to ensure that that additional properties on the action object fail an error.

**Comment by urbantumbleweed** [comment]
I added the tests we talked about that would fail in the event we add a property to the action or payload objects and don't have tests for them.  I used `lodsash.isEqual()` so the properties listed in the test will need to match the order that occurs in the function.  I have alternative approach to this if we must allow the properties to exist in any order but it would be more expensive.

File: src/creators/__tests__/change-directory-error.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+const { isFSA } = require('flux-standard-action');
+const { keys, isEqual } = require('lodash');
+
+const { changeDirectoryError } = require('../');
+const { ERROR } = require('../../constants');
+const { CHANGE_DIRECTORY_FAILURE } = require('../../status-constants');
+
+
+describe(`changeDirectoryError creator`, function(){
+  const actionProperties = [
```
Comment (phated) [inline src/creators/__tests__/change-directory-error.js]: these are needed to be tested.  isFSA already tests this

File: src/creators/__tests__/change-directory-error.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    expect(action.payload.status).toEqual(CHANGE_DIRECTORY_FAILURE);
+    done();
+  });
+
+  it(`returns the original error in payload.error`, function(done){
+    expect(action.payload.error).toEqual(err);
+    done();
+  });
+
+  it(`returns an action object that only has ${actionProperties.length} known ${actionProperties.length === 1 ? 'property' : 'properties'}`, function(done){
```
Comment (phated) [inline src/creators/__tests__/change-directory-error.js]: not needed

File: src/creators/__tests__/change-directory-error.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    expect(isEqual(actionKeys, actionProperties)).toEqual(true);
+    done();
+  });
+
+  it(`returns an action.payload object that only has ${payloadProperties.length} known ${payloadProperties.length === 1 ? 'property' : 'properties'}`, function(done){
```
Comment (phated) [inline src/creators/__tests__/change-directory-error.js]: don't like this variable substitution, `action.payload has only expected properties`

File: src/creators/__tests__/change-directory-error.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+  ];
+  const payloadProperties = [
+    'notification',
+    'status',
+    'error'
+  ];
+  const dirpath = 'change/directory/error/';
+  const err = new Error('Change dir error');
+  const action = changeDirectoryError(dirpath, err);
+  const actionKeys = keys(action);
```
Comment (phated) [inline src/creators/__tests__/change-directory-error.js]: not needed

File: src/creators/__tests__/change-directory-error.js [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    done();
+  });
+
+  it(`returns an action object that only has ${actionProperties.length} known ${actionProperties.length === 1 ? 'property' : 'properties'}`, function(done){
+    expect(isEqual(actionKeys, actionProperties)).toEqual(true);
+    done();
+  });
+
+  it(`returns an action.payload object that only has ${payloadProperties.length} known ${payloadProperties.length === 1 ? 'property' : 'properties'}`, function(done){
+    expect(isEqual(payloadKeys, payloadProperties)).toEqual(true);
```
Comment (phated) [inline src/creators/__tests__/change-directory-error.js]: this should probably just be 3 expects:

``` js
expect(keys(action.payload)).toInclude('notification');
expect(keys(action.payload)).toInclude('status');
expect(keys(action.payload)).toInclude('error');
```

That can be DRYed up a bit, but it makes it more clear

Comment (phated) [inline src/creators/__tests__/change-directory-error.js]: Scratch that, it doesn't test if there are other keys.

Maybe you could do _.omit() and then check length?

Comment (phated) [inline src/creators/__tests__/change-directory-error.js]: Equal fails on an arrays with keys not in the same order

**Comment by urbantumbleweed** [comment]
Code has been refactored and cleaned-up.  I removed the action check, used `_.pick()` for the payload property check, and moved the expectedProps array into the appropriated function.

**Pull request merged**